rust/sealevel/programs/mailbox/src/processor.rs

```diff
@@ -0,0 +1,855 @@
+        OutboxDispatch, OutboxQuery, RecipientInstruction, MAX_MESSAGE_BODY_BYTES, VERSION,
+    },
+};
+
+#[cfg(not(feature = "no-entrypoint"))]
+entrypoint!(process_instruction);
+
+pub fn process_instruction(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    match MailboxIxn::from_instruction_data(instruction_data)? {
+        MailboxIxn::Init(init) => initialize(program_id, accounts, init),
+        MailboxIxn::InboxProcess(process) => inbox_process(program_id, accounts, process),
+        MailboxIxn::InboxSetDefaultModule(ism) => inbox_set_default_ism(program_id, accounts, ism),
+        MailboxIxn::OutboxDispatch(dispatch) => outbox_dispatch(program_id, accounts, dispatch),
+        MailboxIxn::OutboxGetCount(query) => outbox_get_count(program_id, accounts, query),
+        MailboxIxn::OutboxGetLatestCheckpoint(query) => {
+            outbox_get_latest_checkpoint(program_id, accounts, query)
+        }
+        MailboxIxn::OutboxGetRoot(query) => outbox_get_root(program_id, accounts, query),
+    }
+    // .map_err(|err| {
+    //     err.print::<crate::error::Error>();
+    //     err
+    // })
+}
+
+fn initialize(program_id: &Pubkey, accounts: &[AccountInfo], init: Init) -> ProgramResult {
+    // On chain create appears to use realloc which is limited to 1024 byte increments.
+    let mailbox_size = 2048;
+    let accounts_iter = &mut accounts.iter();
+
+    let system_program = next_account_info(accounts_iter)?;
+    if system_program.key != &solana_program::system_program::ID {
+        return Err(ProgramError::InvalidArgument);
+    }
+
+    let payer_account = next_account_info(accounts_iter)?;
+
+    let auth_account = next_account_info(accounts_iter)?;
+    let (auth_key, auth_bump) = Pubkey::find_program_address(
+        &[
+            b"hyperlane",
+            b"-",
+            &init.local_domain.to_le_bytes(),
+            b"-",
+            b"authority",
+        ],
+        program_id,
+    );
+    if &auth_key != auth_account.key {
+        return Err(ProgramError::InvalidArgument);
+    }
+    invoke_signed(
+        &system_instruction::create_account(
+            payer_account.key,
+            auth_account.key,
+            Rent::default().minimum_balance(0),
+            0,
+            program_id,
+        ),
+        &[
+            system_program.clone(),
+            payer_account.clone(),
+            auth_account.clone(),
+        ],
+        &[&[
+            b"hyperlane",
+            b"-",
+            &init.local_domain.to_le_bytes(),
+            b"-",
+            b"authority",
+            &[auth_bump],
+        ]],
+    )?;
+
+    let inbox_account = next_account_info(accounts_iter)?;
+    let (inbox_key, inbox_bump) = Pubkey::find_program_address(
+        &[
+            b"hyperlane",
+            b"-",
+            &init.local_domain.to_le_bytes(),
```

Comment:
now that i think harder about it, probably not necessary to include local domain.

The destination domain might be useful to include in the outbox seeds and the origin domain might be useful to include in the inbox seeds, assuming that would make it easier for off chain programs to examine only messages that are going to / coming from a particular chain. If inbox/outbox have domain in the seeds, so should auth to prevent signing for the wrong domain.

Reply:
the value of the above is questionable however - what do you think?